fvm/src/syscalls/bind.rs

```diff
@@ -148,30 +153,35 @@ macro_rules! impl_bind_syscalls {
                 if mem::size_of::<Ret::Value>() == 0 {
                     // If we're returning a zero-sized "value", we return no value therefore and expect no out pointer.
                     self.func_wrap(module, name, move |mut caller: Caller<'_, InvocationData<K>> $(, $t: $t)*| {
-                        charge_gas_for_exec_units(&mut caller)?;
+                        wasmgas_to_gastracker(&mut caller)?;
+
                         let (mut memory, mut data) = memory_and_data(&mut caller)?;
                         let ctx = Context{kernel: &mut data.kernel, memory: &mut memory};
-                        let result = match syscall(ctx $(, $t)*).into()? {
-                            Ok(_) => {
+                        let out = syscall(ctx $(, $t)*).into();
+
+                        let result = match out {
```

Comment:
I think we can re-combine these lines now.

Reply:
(We'll split them again if we merge #465.)